### PR TITLE
fix esil-paritiy-stuff

### PR DESCRIPTION
**Pull request by condret**
my attempt to fix dat

### CODE REVIEW

File: libr/anal/esil.c [hunk 1]
```diff
@@ -183,15 +183,9 @@ static int esil_internal_carry_check (RAnalEsil *esil, ut8 bit) {
...
-	bits ^= esil->cur & (ut64) ( 1 << 1 );
-	bits ^= esil->cur & (ut64) ( 1 << 2 );
-	bits ^= esil->cur & (ut64) ( 1 << 3 );
-	bits ^= esil->cur & (ut64) ( 1 << 4 );
-	bits ^= esil->cur & (ut64) ( 1 << 5 );
-	bits ^= esil->cur & (ut64) ( 1 << 6 );
-	bits ^= esil->cur & (ut64) ( 1 << 7 );
+	int i, bits = 0;
+	for (i = 0; i < 8; i++)
+		bits ^= ((esil->cur & (1<<i))>>i);
```
Comment (ret2libc) [inline libr/anal/esil.c]: spaces around operators.
```bits ^= ((esil->cur & (1 << i)) >> i);```

**Comment by condret** [comment]
seriously?

**Comment by dequis** [comment]
>paritiy

**Comment by radare** [comment]
Yep. Fix the indent, rebase and merge

**Comment by radare** [comment]
Is this the same as https://github.com/j123123/radare2/commit/4006bfa9db02682f0bb6f8438f978e6732a887ca ?

**Comment by radare** [comment]
What about adding some tests to confirm parity works in esil?

**Comment by j123123** [comment]
Yes, same

**Comment by radare** [comment]
What about using both implementations, just one of them under #if BLACK_MAGIC

File: libr/anal/esil.c (hunk 1 above)
Comment (sushant94) [inline libr/anal/esil.c]: A shorter way would be:
`bits ^= (esil->cur >> i)  & 1;`

**Comment by j123123** [comment]
As you wish. You may also use GCC builtin function https://gcc.gnu.org/onlinedocs/gcc/Other-Builtins.html
`int __builtin_parity (unsigned int x)` with `#ifdef __GNUC__`
should work on GCC, Clang, Intel compiler for Linux. For other compilers, use code without builtin

**Comment by radare** [comment]
I prefer not to use compiler specific stuff, or at least if thats the only way to do it. So if you add an ifdef for supported compilers and use that builtin and provide and alternative for c-only implementation. Its ok to merge. Also, this needs tests to verify nothing breaks in the future

**Comment by jvoisin** [comment]
Please don't use intrinsic, especially for dead-simple code like this; I mean, this is not, at all, a bottleneck.

**Comment by condret** [comment]
will fix ident now and merge

**Pull request merged**